src/commands/importDocuments.ts

```diff
@@ -0,0 +1,134 @@
+
+export async function importDocuments(tree: AzureTreeDataProvider, uris: vscode.Uri[] | undefined, collectionNode: IAzureParentNode<MongoCollectionTreeItem | DocDBCollectionTreeItem> | undefined): Promise<void> {
+    if (!uris) {
+        uris = await askForDocuments();
+    }
+    let ignoredUris: vscode.Uri[] = []; //account for https://github.com/Microsoft/vscode/issues/59782
+    uris = uris.filter((uri) => {
+        if (uri.fsPath.endsWith('.json')) {
+            return true;
+        } else {
+            ignoredUris.push(uri);
+            return false;
+        }
+    });
+    if (ignoredUris.length) {
+        ext.outputChannel.appendLine(`The following selected files are not json: ${ignoredUris.map(uri => uri.fsPath).join(',')}. \nIgnoring these.`);
+        ext.outputChannel.show();
+    }
+    if (!collectionNode) {
+        collectionNode = <IAzureParentNode<MongoCollectionTreeItem | DocDBCollectionTreeItem>>await tree.showNodePicker([MongoCollectionTreeItem.contextValue, DocDBCollectionTreeItem.contextValue]);
+    }
+
+    const documents = await vscode.window.withProgress(
+        {
+            location: vscode.ProgressLocation.Notification,
+            title: "Import documents..."
+        },
+        (_progress) => parseDocumentsForErrors(uris)
+    );
+
+    let result: string;
+    if (collectionNode.treeItem instanceof MongoCollectionTreeItem) {
+        const collectionTreeItem = <MongoCollectionTreeItem>collectionNode.treeItem;
+        result = await collectionTreeItem.executeCommand('insertMany', [JSON.stringify(documents)]);
+    } else {
+        result = await insertDocumentsIntoDocdb(<IAzureParentNode<DocDBCollectionTreeItem>>collectionNode, documents, uris);
+    }
+    await collectionNode.refresh();
+    await vscode.window.showInformationMessage(result);
+}
+
+async function askForDocuments(): Promise<vscode.Uri[]> {
+    return await ext.ui.showOpenDialog({
+        canSelectMany: true,
+        openLabel: "Import",
+        filters: {
+            "JSON": ["json"]
+        },
+        defaultUri: vscode.Uri.file(vscode.workspace.rootPath)
+    });
+}
+
+// tslint:disable-next-line:no-any
+async function parseDocumentsForErrors(nodes: vscode.Uri[]): Promise<any[]> {
+    const parseResult = await parseDocuments(nodes);
+    const documents = parseResult[0];
+    const hasErrors: boolean = parseResult[1];
+    if (hasErrors) {
+        throw new Error(`Errors found in some documents.\nPlease see the output, fix these and try again.`);
+    }
+    return documents;
+}
+
+// tslint:disable-next-line:no-any
+async function parseDocuments(uris: vscode.Uri[]): Promise<[any[], boolean]> {
+    let documents = [];
+    let errors = {};
+    let errorFoundFlag: boolean = false;
+    for (let uri of uris) {
+        const text: string = await fse.readFile(uri.fsPath, 'utf-8');
+        let parsed;
+        try {
+            parsed = JSON.parse(text);
+        } catch (e) {
+            if (!errorFoundFlag) {
```

Comment:
You never set `errorFoundFlag` to true, so this gets output multiple times

```
Errors found in documents listed below. Please fix these.
/Users/ericjizba/TestRepos/py1/doc1.json:
Unexpected end of JSON input
Errors found in documents listed below. Please fix these.
/Users/ericjizba/TestRepos/py1/doc2.json:
Unexpected end of JSON input
```

I would prefer if the output looked like this:
```
Errors found in documents listed below. Please fix these.
/Users/ericjizba/TestRepos/py1/doc1.json: Unexpected end of JSON input
/Users/ericjizba/TestRepos/py1/doc2.json: Unexpected end of JSON input
```